src/main/java/input_feed/serial/JSONToSerialR1CS.java

```diff
@@ -0,0 +1,124 @@
+package input_feed.serial;
+
+import algebra.fields.AbstractFieldElementExpanded;
+import algebra.fields.Fp;
+import algebra.fields.abstractfieldparameters.AbstractFpParameters;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import relations.objects.*;
+import relations.r1cs.R1CSRelation;
+import scala.Tuple2;
+
+import java.io.FileReader;
+import java.lang.reflect.Field;
+
+public class JSONToSerialR1CS<FieldT extends AbstractFieldElementExpanded<FieldT>>
```

Comment:
Is there a reason that this file cannot share logic with the Distributed JSON reader? If I understand correctly the only difference is the one is called via a sparkContext.parallelize but I wonder if the logic for e.g. parsing a single line could be captured in a single method that is shared across the serial and distributed implementation.

Same question applies for serial/distributed file reader